subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/SkipCachedTaskExecuter.java

```diff
@@ -115,18 +135,49 @@ public void execute(final TaskInternal task, TaskStateInternal state, TaskExecut
         }
     }
 
-    private static SortedSet<ResolvedTaskOutputFilePropertySpec> resolveProperties(ImmutableSortedSet<? extends TaskOutputFilePropertySpec> properties) {
-        ImmutableSortedSet.Builder<ResolvedTaskOutputFilePropertySpec> builder = ImmutableSortedSet.naturalOrder();
+    private static SortedSet<CacheableTree> resolveProperties(ImmutableSortedSet<? extends TaskOutputFilePropertySpec> properties) {
+        ImmutableSortedSet.Builder<CacheableTree> builder = ImmutableSortedSet.naturalOrder();
         for (TaskOutputFilePropertySpec property : properties) {
             // At this point we assume that the task only has cacheable properties,
             // otherwise caching would have been disabled by now
-            CacheableTaskOutputFilePropertySpec cacheableProperty = (CacheableTaskOutputFilePropertySpec) property;
-            builder.add(new ResolvedTaskOutputFilePropertySpec(
```

Comment:
ResolvedTaskOutputFilePropertySpec can be removed now.

Reply:
Removed.